### DISCUSSION TITLE
Pop sound on boot in SLG2?

### DISCUSSION THREAD

**Discussion by Opposite34** [linux-surface/linux-surface#1524] [General]
Hi, so I have been experiencing a really loud and feel-able pop sound on boot for SLG2. It seems to come from the middle of the keyboard (not sure exactly what hardware lies there). I have tried disabling the intel-snd-hda power saving option and blacklisted pcspkr (in `modprobe.d`, but to no avail, said pop sound still comes when boot. Of note that it was fixed for a while somehow but came back after system + kernel update, said issue also appeared in the ootb non-surface kernel as well.

Currently on:
- Surface Laptop Go 2 (SLG2)
- Debian Testing, with Linux 6.10.3-surface-1
- GNOME 46

Should this also be something to concerned about hardware-wise? (since it doesn't seem to be the pcspkr module, so it's not the pc beep-on-boot sound, I'm a bit concerned it something with the hardware, and if it can break). Even if not, if someone knows how to disable it please inform still, as it is quite loud and annoying.

PS: I hope this is the right place for it, as Issues should be for issues on the kernel itself not surface hardware on linux, but if it isn't please tell me.

**Comment by Opposite34** [comment]
This randomly disappear after dual booting into windows and then back to linux... which is really weird... My guess is maybe the pcspkr didn't get blacklisted until booting another os(????), but that doesn't sound right. Another guess would be if Microsoft recently have a UEFI updates that fixes this somehow(?). Either way, I'll update if this happens in future kernel updates in case other people are also experiencing this.

**Comment by Opposite34** [comment]
This starts happening again after the latest kernel update. I am so confused why it happens after a kernel update.

**Comment by Opposite34** [comment]
Update to this, this does seem to happen still. The pop sound comes out from middle of the keyboard. I don't know what's there but it is very noticeable.

I noticed that updating the initramfs *even* without blacklisting `pcspkr` and the `snd-hda-intel` power saving configs might have make this go away, but this does mean it is some modules possibly, just not the ones in modprobe.d (that one only has the microcode thingy that I assumed is autogenerated by `linux-surface` in it, and I doubt that is affecting this pop sound somehow???).

Next kernel update I'll try to manually update the initramfs without those configs before rebooting and see if the pop sound happens on boot or not as I have not tried this, if this is needed I don't mind doing it manually after kernel updates on my end.

Edited: I updated to 6.18.5 just now when I was previously in 6.18.4. My theory was very wrong. updating initramfs doesn't do anything to prevent the pop. I think it just disappeared on its own the last time then... like random after using the laptop for a bit.... yeah this is weird...

**Reply by Opposite34** [reply]
Okay so I just realized I did not disable fast startup on windows side (still has it just as a fallback even though I rare log into it.).
Did that just now, and it seems to be gone(?). Let's see if it stays like this the next kernel update. But I feel like this is most liekly some kind of power states issue.

Edited: nevermind, it's still producing the pop sounds....

**Comment by ruslanbay** [comment]
Not sure, but this might help: https://github.com/linux-surface/linux-surface/wiki/Known-Issues-and-FAQ#speaker-noise-when-headphones-plugged-in
